File: test_orbit_provider.py
import httpx
import json

# URL
url = "https://api.orbit-provider.com/v1/chat/completions"

# Headers
headers = {
    "Authorization": "Bearer sk-orbit-6***3cba",
    "Content-Type": "application/json"
}

# JSON payload
payload = {
    "model": "claude-sonnet-4-5-20250929",
    "messages": [
        {"role": "user", "content": "Hello!"}
    ]
}

print("Testing Orbit Provider API...")
print(f"URL: {url}")
print(f"Headers: {headers}")
print(f"Payload: {payload}")

try:
    # Sending the POST request
    response = httpx.post(url, headers=headers, json=payload, timeout=30.0)

    # Print status code and response
    print(f"\nStatus Code: {response.status_code}")
    
    try:
        response_json = response.json()
        print("Response JSON:")
        print(json.dumps(response_json, indent=2))
    except:
        print("Response Text:")
        print(response.text)

except Exception as e:
    print(f"\nError occurred: {str(e)}")

File: test_orbit_provider.js
import fetch from 'node-fetch';

// Test the Orbit Provider API
async function testOrbitProviderAPI() {
    console.log('Testing Orbit Provider API...\n');
    
    const url = 'https://api.orbit-provider.com/v1/chat/completions';
    const headers = {
        'Authorization': 'Bearer sk-orbit-6***3cba',  // Try Authorization header as alternative
        'Content-Type': 'application/json'
    };
    
    const payload = {
        "model": "claude-sonnet-4-5-20250929",
        "messages": [
            {"role": "user", "content": "Hello!"}
        ]
    };

    try {
        console.log('URL:', url);
        console.log('Headers:', headers);
        console.log('Payload:', JSON.stringify(payload, null, 2));
        
        const response = await fetch(url, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(payload)
        });

        console.log('\nStatus Code:', response.status);
        
        const responseBody = await response.text();
        console.log('Response Body:');
        
        try {
            const jsonData = JSON.parse(responseBody);
            console.log(JSON.stringify(jsonData, null, 2));
        } catch (e) {
            console.log(responseBody);
        }

        if (response.ok) {
            console.log('\n✅ Orbit Provider API test successful!');
        } else {
            console.log(`\n❌ API returned error status: ${response.status}`);
        }

    } catch (error) {
        console.error('\n❌ Request failed:', error.message);
    }
}

// Run the test
await testOrbitProviderAPI();